docs/reference/ImageDraw.rst

```diff
@@ -476,6 +496,164 @@ Methods
 
                      .. versionadded:: 6.2.0
 
+.. py:method:: ImageDraw.textlength(text, font=None, direction=None, features=None, language=None)
+
+    Returns length (in pixels with 1/64 precision) of given text if rendered
+    in font with provided direction, features, and language.
+
+    This is the amount by which following text should be offset.
+    Text bounding box may extend past the length in some fonts,
+    e.g. when using italics or accents.
+
+    The result is returned as a float; it is a whole number if using basic layout.
+
+    This method uses :meth:`.FreeTypeFont.getlength` internally and falls back
+    on :meth:`textlength` for non-TrueType fonts.
```

Comment:
Are details of the internal implementation necessary in these docs?

Reply:
I think it is an interesting detail (i.e. this function is only really supported for TrueType fonts), but probably not interesting enough to keep this comment. Will remove.